File: recruitment-be/src/main/java/com/psybergate/recruitment/marking/ai/AiMarkingService.java
package com.psybergate.recruitment.marking.ai;

import com.psybergate.recruitment.marking.ai.dto.AiMarkingSuggestionResponse;
import java.util.UUID;

public interface AiMarkingService {
    AiMarkingSuggestionResponse generateSuggestion(UUID submissionId, UUID questionId);

    AiMarkingSuggestionResponse getSuggestion(UUID submissionId, UUID questionId);
}

File: recruitment-fe/src/app/core/ai-marking/ai-marking.service.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AiMarkingSuggestionResponse } from './ai-marking.model';

@Injectable({ providedIn: 'root' })
export class AiMarkingService {
  private readonly http = inject(HttpClient);

  getSuggestion(submissionId: string, questionId: string): Observable<AiMarkingSuggestionResponse> {
    return this.http.get<AiMarkingSuggestionResponse>(
      `/api/submissions/${submissionId}/questions/${questionId}/ai-suggestion`,
    );
  }

  generateSuggestion(submissionId: string, questionId: string): Observable<AiMarkingSuggestionResponse> {
    return this.http.post<AiMarkingSuggestionResponse>(
      `/api/submissions/${submissionId}/questions/${questionId}/ai-suggestion`,
      {},
    );
  }
}
